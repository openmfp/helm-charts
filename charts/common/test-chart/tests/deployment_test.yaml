suite: deployment
templates:
  - deployment.yaml
values:
  - ../values.yaml
chart:
  namespace: default
  appVersion: 0.1.0
tests:
  - it: test tag default
    asserts:
      - matchSnapshot: {}
  - it: test tag override
    set:
      image:
        tag: 1.0.0
    asserts:
      - matchSnapshot: {}
  - it: test name override
    set:
      image:
        name: my-image
    asserts:
      - matchSnapshot: {}
  - it: test sentry enabled
    set:
      image:
        name: my-image
      global:
        sentry:
          enabled: true
      environment: "test"
      sentry:
        environment: "test"
    asserts:
      - matchSnapshot: {}
  - it: test extraEnvs
    set:
      extraEnvs:
        - name: MY_ENV
          value: my-value
      extraVolumes:
        - name: my-volume
          emptyDir: {}
    asserts:
      - matchSnapshot: {}
  - it: test global region
    set:
      global:
        region: us-east-1
    asserts:
      - matchSnapshot: {}
