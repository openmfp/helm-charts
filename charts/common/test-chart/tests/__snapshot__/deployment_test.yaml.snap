test extraEnvs:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-test-chart
      template:
        spec:
          containers:
            - env:
                - name: LOG_LEVEL
                  value: info
                - name: MY_ENV
                  value: my-value
              image: test-chart:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: manager
              ports:
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 100m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
          securityContext:
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          terminationGracePeriodSeconds: 10
          volumes:
            - emptyDir: {}
              name: my-volume
test name override:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-test-chart
      template:
        spec:
          containers:
            - env:
                - name: LOG_LEVEL
                  value: info
              image: my-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: manager
              ports:
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 100m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
          securityContext:
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          terminationGracePeriodSeconds: 10
          volumes: null
test sentry enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-test-chart
      template:
        spec:
          containers:
            - env:
                - name: LOG_LEVEL
                  value: info
                - name: REGION
                  value: null
                - name: ENVIRONMENT
                  value: test
                - name: SENTRY_ENVIRONMENT
                  value: test
                - name: IMAGE_TAG
                  value: 0.1.0
                - name: IMAGE_NAME
                  value: my-image
              image: my-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: manager
              ports:
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 100m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
          securityContext:
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          terminationGracePeriodSeconds: 10
          volumes: null
test tag default:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-test-chart
      template:
        spec:
          containers:
            - env:
                - name: LOG_LEVEL
                  value: info
              image: test-chart:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: manager
              ports:
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 100m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
          securityContext:
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          terminationGracePeriodSeconds: 10
          volumes: null
test tag override:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME-test-chart
      template:
        spec:
          containers:
            - env:
                - name: LOG_LEVEL
                  value: info
              image: test-chart:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8081
                periodSeconds: 10
              name: manager
              ports:
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 100m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
          securityContext:
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          terminationGracePeriodSeconds: 10
          volumes: null
